src/Job.php

```diff
@@ -62,16 +63,19 @@ public function delete()
      */
     public function release($delay = 0)
     {
-        if ($delay) {
-            throw new \LogicException('To be implemented');
-        }
+        parent::release($delay);
 
         $requeueMessage = clone $this->psrMessage;
         $requeueMessage->setProperty('x-attempts', $this->attempts() + 1);
 
-        $this->psrContext->createProducer()->send($this->psrConsumer->getQueue(), $requeueMessage);
+        $producer = $this->psrContext->createProducer();
+
+        if ($producer instanceof DelayStrategyAware && $producer->delayable()) {
```

Comment:
what is `$producer->delayable()`? There is no such method? 

`$producer instanceof DelayStrategyAware` makes sense only for AMQP transports. 

It'd be better to try always setDeliveryDelay and catch an exception DeliveryDelayNotSupported (or something like this)>